template-extension.js

```diff
@@ -12,6 +12,15 @@ Meteor.startup(function () {
     function defineHook(type) {
       // see if there's an existing callback set directly on the template instance
       var orig = template[type];
+
+      // Basically scraping callbacks set directly on instance and saving
+      // in templateHooks
+      if (typeof orig === 'function') {
+        var name = parseName(template.viewName);
+        templateHooks[type][name] = templateHooks[type][name] || [];
+        templateHooks[type][name].push(orig);
+      }
```

Comment:
I think this is a good way to do it, but then we'll need to remove the "call the existing callback if there was one" block below, right? Otherwise I think the existing callbacks will be called twice.